### PR TITLE
Ignore page cache for docker memory usage

### PR DESCRIPTION
**Pull request by solarkennedy**
No testing yet.

### CODE REVIEW

**Comment by solarkennedy** [comment]
cc @chriskuehl

File: src/fullerite/collector/docker_stats.go [hunk 1]
```diff
@@ -189,7 +189,9 @@ func (d *DockerStats) extractMetrics(container *docker.Container, stats *docker.
 
 // buildMetrics creates the actual metrics for the given container.
 func (d DockerStats) buildMetrics(container *docker.Container, containerStats *docker.Stats, cpuPercentage float64) []metric.Metric {
+	mem := containerStats.MemoryStats.Usage - containerStats.MemoryStats.Stats.Cache
 	ret := []metric.Metric{
+		buildDockerMetric("DockerMemoryUsed", metric.Gauge, float64(mem)),
 		buildDockerMetric("DockerMemoryUsed", metric.Gauge, float64(containerStats.MemoryStats.Usage)),
```
Comment (chriskuehl) [inline src/fullerite/collector/docker_stats.go]: should this delete the metric below it with the same name? or rename one of them?

**Comment by chriskuehl** [comment]
cc @magicmark

**Comment by solarkennedy** [comment]
Related to https://github.com/aws/amazon-ecs-agent/pull/582 and https://github.com/moby/moby/issues/10824

**Comment by dichiarafrancesco** [comment]
lg2m. please fix tests

**Comment by solarkennedy** [comment]
I could not reproduce the broken tests on my devbox, but they failed consistently in travis.

I "Fixed" it by upgrading the version of golang on travis to match my devbox.

**Pull request merged**